dali/pipeline/operators/reader/nvdecoder/imgproc.cu

```diff
@@ -24,139 +24,171 @@ namespace {
-    } else {
-        auto& m = yuv2rgb_mat;
-        r = clip(y*m[0] + u*m[1] + v*m[2], 255.0);
-        g = clip(y*m[3] + u*m[4] + v*m[5], 255.0);
-        b = clip(y*m[6] + u*m[7] + v*m[8], 255.0);
-    }
-
-    rgb[0] = convert<RGB_T>(r);
-    rgb[stride] = convert<RGB_T>(g);
-    rgb[stride*2] = convert<RGB_T>(b);
+  auto y = (static_cast<float>(yuv.y) - 16.0f/255.0f);
+  auto u = (static_cast<float>(yuv.u) - 128.0f/255.0f);
+  auto v = (static_cast<float>(yuv.v) - 128.0f/255.0f);
+
+
+  float r, g, b;
+  if (Normalized) {
+    auto& m = yuv2rgb_mat_norm;
+    r = clip(y*m[0] + u*m[1] + v*m[2], 1.0f);
+    g = clip(y*m[3] + u*m[4] + v*m[5], 1.0f);
+    b = clip(y*m[6] + u*m[7] + v*m[8], 1.0f);
+  } else {
+    auto& m = yuv2rgb_mat;
+    r = clip(y*m[0] + u*m[1] + v*m[2], 255.0f);
+    g = clip(y*m[3] + u*m[4] + v*m[5], 255.0f);
+    b = clip(y*m[6] + u*m[7] + v*m[8], 255.0f);
+  }
+
+  rgb[0] = convert<RGB_T>(r);
+  rgb[stride] = convert<RGB_T>(g);
+  rgb[stride*2] = convert<RGB_T>(b);
 }
 
-template<typename T, bool Normalized = false>
+template<typename T, bool Normalized = false, bool RGB = true>
 __global__ void process_frame_kernel(
-    cudaTextureObject_t luma, cudaTextureObject_t chroma,
-    T* dst, int index,
-    float fx, float fy,
-    int dst_width, int dst_height, int c) {
-
-    const int dst_x = blockIdx.x * blockDim.x + threadIdx.x;
-    const int dst_y = blockIdx.y * blockDim.y + threadIdx.y;
-
-    if (dst_x >= dst_width || dst_y >= dst_height)
-        return;
-
-    auto src_x = 0.0f;
-    src_x = static_cast<float>(dst_x) * fx;
-    auto src_y = static_cast<float>(dst_y) * fy;
-
-
-
-
-    // TODO(spanev) something less hacky here, why 4:2:0 fails on this edge?
-    float shift = (dst_x == dst_width - 1) ? 0 : 0.5f;
-
-    // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#tex2d-object
-    yuv<float> yuv;
-    yuv.y = tex2D<float>(luma, src_x + shift, src_y + shift);
-    auto uv = tex2D<float2>(chroma, (src_x / 2) + shift, (src_y / 2) + shift);
-    yuv.u = uv.x;
-    yuv.v = uv.y;
-
-    auto* out = &dst[(dst_x + dst_y * dst_width) * c];
-
-    size_t stride = 1;
-    yuv2rgb<float, float, Normalized>(yuv, out, stride);
+  cudaTextureObject_t luma, cudaTextureObject_t chroma,
+  T* dst, int index,
+  float fx, float fy,
+  int dst_width, int dst_height, int c) {
+  const int dst_x = blockIdx.x * blockDim.x + threadIdx.x;
+  const int dst_y = blockIdx.y * blockDim.y + threadIdx.y;
+
+  if (dst_x >= dst_width || dst_y >= dst_height)
+      return;
+
+  auto src_x = 0.0f;
+  src_x = static_cast<float>(dst_x) * fx;
+  auto src_y = static_cast<float>(dst_y) * fy;
+
+  // TODO(spanev) something less hacky here, why 4:2:0 fails on this edge?
+  float shift = (dst_x == dst_width - 1) ? 0 : 0.5f;
+
+  // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#tex2d-object
+  yuv<float> yuv;
+  yuv.y = tex2D<float>(luma, src_x + shift, src_y + shift);
```

Comment:
There was some info from @ptrendx that texture interpolation uses low precision - could you check if that may affect us?

Reply:
I could change the CUDA filter mode to cudaFilterModePoint for the texture also (right now its hardcoded to bilinear)
I think that this would be a matter for another discussion and another PR